File: BussinessObjects/Models/Review.cs
namespace BussinessObjects.Models;

public class Review
{
    public int ReviewId { get; set; }

    public int BookingId { get; set; }

    public int UserId { get; set; }

    public int HelperId { get; set; }

    public int Rating { get; set; }

    public string? Comment { get; set; }

    public DateTime? ReviewDate { get; set; }

    public bool? IsEdited { get; set; }

    public DateTime? LastEditedDate { get; set; }

    public virtual Booking Booking { get; set; } = null!;

    public virtual Helper Helper { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: src/types/review.ts
// Review Types

export interface Review {
  reviewId: number;
  bookingId: number;
  helperId: number;
  userId: number;
  rating: number;
  comment: string;
  reviewDate: string;
}

export interface CreateReviewRequest {
  bookingId: number;
  helperId: number;
  rating: number;
  comment: string;
}

export interface UpdateReviewRequest {
  rating: number;
  comment: string;
}

export interface ReviewResponse {
  success: boolean;
  statusCode: number;
  message: string;
  data: Review | Review[];
  timestamp: string;
  requestId: string;
}

export interface ReviewFormData {
  rating: number;
  comment: string;
} 